### ISSUE TITLE
Unable to call extension-contributed commands in preLaunchTask without adding to task.json

### ISSUE THREAD

**Issue by DanTup** [Microsoft/vscode#61588]
Assuming I have an extension that contributes some commands like:

```ts
export class PubTaskProvider implements vs.TaskProvider {
	public provideTasks(token?: vs.CancellationToken): vs.ProviderResult<vs.Task[]> {
		return [
			new vs.Task({ type: "pub", command: "get", taskName: "pub get" }, vs.workspace.workspaceFolders[0], "get", "pub", new vs.ShellExecution("pub get"), []),
		];
	}
```

It seems that it's not possible to call from a `preLaunchTask` without first adding to `task.json` (for ex. by running the `Configure Task` command).

If I don't need to configure anything, it seems strange that it works if I just run `Configure` and then make zero changes. It would be great if I could just run the contributed tasks with their default config without having to create a task.json (similar to how we can run without `launch.json`).

**Comment by dbaeumer** [comment]
@DanTup you are better using 
```json
{
   "type": "pub",
   "command": "get"
}
```

This makes it locale independent.

**Comment by dbaeumer** [comment]
You can put that object into the `prelaunchTask` value assuming that the `command` property is a property that the pub task defines in its task identifier. For `npm` tasks which are contributed by an extension the discriminting property is the `script`. So such a task can be referenced like this
```json
{
    "type": "npm",
    "script": "....."
}
```

**Comment by DanTup** [comment]
Gotcha! I'm actually trying to make it as simple as possible for my users. I'm providing a task with the hope that they just make a tiny change to `launch.json` if they want this functionality. The string is much simpler - could you elaborate on the issues with locales that might affect it?

**Comment by dbaeumer** [comment]
`pub: get` is a UI label and will be formatted in French like this: `pub : get`. So if a users inserts that then this will very likely be fine since on a French machine the user will see `pub : get` and put it that way into the launch.json. The literal reference is locale independent and should be used when things get inserted automatically.

**Comment by DanTup** [comment]
Makes sense; I'll stick to the literal then - it's not too bad and most people will copy/paste (and we may ultimately inject it in the resolve method based on settings or something). Thanks!
